packages/ckeditor5-ui/src/colorpicker/colorpickerview.ts

```diff
@@ -104,13 +104,19 @@ export default class ColorPickerView extends View {
 		} );
 
 		// Sets color in the picker if color was updated.
-		this.on( 'change:color', () => {
-			const convertedColor = convertColor( this.color, this._format );
+		this.on( 'set:color', ( evt, propertyName, newValue, oldValue ) => {
+			const convertedColor = convertColor( newValue, this._format );
+
+			if ( convertedColor != oldValue ) {
```

Comment:
Since in `else` the values are equal - the later `evt.return = oldValue;` is equivalent of `evt.return = convertedColor;`. So in 5dda2d7bbf I simplified this path.